modules/geolocation/geolocation.py

```diff
@@ -96,6 +95,42 @@ def gather_point_pairs(self):
             pixelGeoPairs = np.concatenate((pixelGeoPairs, [pair]))
 
         return pixelGeoPairs
+    
+    def get_non_collinear_points(self, coordinatesArray):
+        """
+        Returns a list of four coordinates from an input array that are not collinear to one another.
+
+        Parameters
+        ----------
+        coordinatesArray : np.ndarray
+            Array with dimensions ( , 2), containing a list of coordinates
+
+        Returns
+        -------
+        np.ndarray
+            Array with dimensions (4, 2), containing a list of coordinates that are non-collinear,
+            or an empty list if none were found in the input array
+        """
+        collinearPoints = np.zeros(shape=(4, 2))
+
+        if len(coordinatesArray) < 4:
+            return collinearPoints
+
+        for i in range(0, len(coordinatesArray)-3):
+            x1, y1 = coordinatesArray[i]
+            x2, y2 = coordinatesArray[i+1]
+            x3, y3 = coordinatesArray[i+2]
+            x4, y4 = coordinatesArray[i+3]
+
+            if (0.5*(x1*(y2-y3)+x2*(y3-y1)+x3*(y1-y2)) != 0 and
+                0.5*(x2*(y3-y4)+x3*(y4-y2)+x4*(y2-y3)) != 0 and
+                0.5*(x3*(y4-y1)+x4*(y1-y3)+x1*(y3-y4)) != 0 and
+                0.5*(x4*(y1-y2)+x1*(y2-y4)+x2*(y4-y1)) != 0):
+                
```

Comment:
Likewise, lines 125-129 should be pulled out into their own functions, and implemented like:

```
def checkColinear(x, y):

all_colinear = False

for i in range(0, 4): 

  i_mod = i % 5
  i_1_mod = (i+1) % 5
  i_2_mod = (i+2) % 5
  all_colinear = all_colinear AND ( 0.5 * (x[i_mod]*(y[i_1_mod] - y[i_2_mod])+x[i_1_mod]*(y[i_2_mod] - y[i_mod]) + x[i_2_mod]*(y[i_mod] - y[i_1_mod]) != 0)

return all_colinear
```
Feel free to change up or challenge the implementation in both my recommendations

Reply:
I think this does make sense for readability; do you think I should define the function within get_non_collinear_points() or as a new class member?